core/capabilities/syncer.go

```diff
@@ -56,26 +71,486 @@ var defaultStreamConfig = p2ptypes.StreamConfig{
+
+	err = s.dispatcher.SetReceiver(
+		fullCapID,
+		fmt.Sprint(don.Id),
+		capability,
+	)
+	if err != nil {
+		return err
+	}
+	err = capability.Start(ctx)
+	if err != nil {
+		return err
+	}
+	s.subServices = append(s.subServices, capability)
+	return nil
+}
+
+var (
+	defaultTargetRequestTimeout = time.Minute
+)
+
+func (s *registrySyncer) enableExternalAccess(ctx context.Context, myPeerID p2ptypes.PeerID, don kcr.CapabilityRegistryDONInfo, state state, remoteWorkflowDONs []kcr.CapabilityRegistryDONInfo) error {
+	idsToDONs := map[string]capabilities.DON{}
+	for _, d := range remoteWorkflowDONs {
+		idsToDONs[fmt.Sprint(d.Id)] = *toDONInfo(d)
+	}
+
+	for _, c := range don.CapabilityConfigurations {
+		capability, ok := state.IDsToCapabilities[c.CapabilityId]
+		if !ok {
+			return fmt.Errorf("could not find capability matching id %s", c.CapabilityId)
+		}
+
+		switch toCapabilityType(capability.CapabilityType) {
+		case capabilities.CapabilityTypeTrigger:
+			newTriggerPublisher := func(capability capabilities.BaseCapability, info capabilities.CapabilityInfo) (receiverService, error) {
+				cfg := &remotetypes.RemoteTriggerConfig{}
+				cfg.ApplyDefaults()
+				err := proto.Unmarshal(c.Config, cfg)
+				if err != nil {
+					return nil, err
+				}
+				publisher := remote.NewTriggerPublisher(
+					cfg,
+					capability.(capabilities.TriggerCapability),
+					info,
+					*toDONInfo(don),
+					idsToDONs,
+					s.dispatcher,
+					s.lggr,
+				)
+				return publisher, nil
+			}
+
+			err := s.addReceiver(ctx, capability, don, newTriggerPublisher)
+			if err != nil {
+				return fmt.Errorf("failed to add server-side receiver: %w", err)
+			}
+		case capabilities.CapabilityTypeAction:
+			s.lggr.Warn("no remote client configured for capability type action, skipping configuration")
+		case capabilities.CapabilityTypeConsensus:
+			s.lggr.Warn("no remote client configured for capability type consensus, skipping configuration")
+		case capabilities.CapabilityTypeTarget:
+			newTargetServer := func(capability capabilities.BaseCapability, info capabilities.CapabilityInfo) (receiverService, error) {
+				return target.NewServer(
+					myPeerID,
+					capability.(capabilities.TargetCapability),
+					info,
+					*toDONInfo(don),
+					idsToDONs,
+					s.dispatcher,
+					defaultTargetRequestTimeout,
+					s.lggr,
+				), nil
+			}
+
+			err := s.addReceiver(ctx, capability, don, newTargetServer)
+			if err != nil {
+				return fmt.Errorf("failed to add server-side receiver: %w", err)
+			}
+		default:
+			s.lggr.Warnf("unknown capability type, skipping configuration: %+v", capability)
+		}
+	}
+	return nil
+}
+
+type receiverService interface {
+	services.Service
+	remotetypes.Receiver
+}
+
+func (s *registrySyncer) addReceiver(ctx context.Context, capability kcr.CapabilityRegistryCapability, don kcr.CapabilityRegistryDONInfo, newReceiverFn func(capability capabilities.BaseCapability, info capabilities.CapabilityInfo) (receiverService, error)) error {
```

Comment:
(Repeating myself) This should be a method on the LocalCapabilitiesRegistry (or Launcher), not Syncer.